(SUP-372) PDK conversion

jarretlavallee opened the pull request:
Prior to this PR the module used one off spec and unit tests. This
commit converts the repo to use the PDK with spec tests. All validation
warnings and errors have been remediated as a part of this commit.

The travis jobs should still be functional and all spec/rubocop test should now complete.

suckatrash commented:
Sorry @jarretlavallee, this is on my to-do list but I haven't gotten to it yet

suckatrash commented on manifests/repos.pp:
```diff
@@ -12,7 +17,6 @@
           gpgcheck => 1,
           baseurl  => 'https://repos.influxdata.com/rhel/$releasever/$basearch/stable',
           gpgkey   => 'https://repos.influxdata.com/influxdb.key',
-          before   => Package['influxdb'],
```
What was the reason for removing the before params here?  Does auto-require handle this?

jarretlavallee replied:
I noticed that we had redundant `before` and `require` metaparameters. We have a `require` on the whole repo class in https://github.com/puppetlabs/puppetlabs-pe_metrics_dashboard/blob/master/manifests/install.pp#L37, so the `before` seemed unnecessary as it would already be in the dependency list. I removed the `before` statements to simplify the respec testing and the dependency tree.

jarretlavallee commented:
Good point about the dev branch. Should we have a `release`, `master`, and `development` branch? What kind of acceptance testing should we do to promote between the branches? I can write those out as a part of our test suite. Right now all I do is spin up an ubuntu and centos container and ensure that both services start and listen on their ports. I can probably do some more advanced querying against influx and grafana.

jarretlavallee commented:
Rebased and targeted onto the development branch. I also added a few simple acceptance spec tests to validate the InfluxDB and Grafana services over the API.
